src/jqLite.js

```diff
@@ -169,20 +179,39 @@ var HTML_REGEXP = /<|&#?\w+;/;
 var TAG_NAME_REGEXP = /<([\w:-]+)/;
 var XHTML_TAG_REGEXP = /<(?!area|br|col|embed|hr|img|input|link|meta|param)(([\w:-]+)[^>]*)\/>/gi;
 
+// Table parts need to be wrapped with `<table>` or they're
+// stripped to their contents when put in a div.
+// XHTML parsers do not magically insert elements in the
+// same way that tag soup parsers do, so we cannot shorten
+// this by omitting <tbody> or other required elements.
 var wrapMap = {
-  'option': [1, '<select multiple="multiple">', '</select>'],
-
-  'thead': [1, '<table>', '</table>'],
-  'col': [2, '<table><colgroup>', '</colgroup></table>'],
-  'tr': [2, '<table><tbody>', '</tbody></table>'],
-  'td': [3, '<table><tbody><tr>', '</tr></tbody></table>'],
-  '_default': [0, '', '']
+  thead: ['table'],
+  col: ['colgroup', 'table'],
+  tr: ['tbody', 'table'],
+  td: ['tr', 'tbody', 'table']
 };
 
-wrapMap.optgroup = wrapMap.option;
 wrapMap.tbody = wrapMap.tfoot = wrapMap.colgroup = wrapMap.caption = wrapMap.thead;
 wrapMap.th = wrapMap.td;
 
+// Support: IE <10 only
+// IE 9 requires an option wrapper & it needs to have the whole table structure
+// set up up front; assigning `"<td></td>"` to `tr.innerHTML` doesn't work, etc.
+var wrapMapIE9 = {
+  option: [1, '<select multiple="multiple">', '</select>'],
+
+  thead: [1, '<table>', '</table>'],
+  col: [2, '<table><colgroup>', '</colgroup></table>'],
+  tr: [2, '<table><tbody>', '</tbody></table>'],
+  td: [3, '<table><tbody><tr>', '</tr></tbody></table>'],
+
+  _default: [0, '', '']
+};
+
+wrapMapIE9.tbody = wrapMapIE9.tfoot = wrapMapIE9.colgroup =
+  wrapMapIE9.caption = wrapMapIE9.thead;
+wrapMapIE9.th = wrapMapIE9.td;
+wrapMapIE9.optgroup = wrapMapIE9.option;
```

Comment:
Seems like a nice idea!

Reply:
Two issues in your version:
1. The `optgroup` assignment is missing.
1. The opening tags are constructed in a wrong order; they need to be in reverse.

Fixing them both reduces the size gains to `118` bytes but that's still `289` extra bytes instead of `407` so a nice win.